stl/inc/complex

```diff
@@ -990,9 +990,28 @@ protected:
                 this->_Val[_IM] = (this->_Val[_IM] - this->_Val[_RE] * _Wr) / _Wd;
                 this->_Val[_RE] = _Tmp;
             }
-        } else if (_Rightimag == 0) { // set NaN result
-            this->_Val[_RE] = _Myctraits::_Nanv();
-            this->_Val[_IM] = this->_Val[_RE];
+        } else if (_Rightimag == 0) { // _Right.real() = 0 && _Right.imag() = 0
+            if (this->_Val[_RE] > 0) {
+                this->_Val[_RE] = _Myctraits::_Infv();
+            } else if (this->_Val[_RE] == 0) {
+                this->_Val[_RE] = -_Myctraits::_Nanv();
+            } else if (this->_Val[_RE] < 0) {
+                this->_Val[_RE] = -_Myctraits::_Infv();
+            } else {
+                // this->_Val[_RE] is NAN

```

Comment:
I suspect (although I'm not sure) that we should test for nan with `_Isnan`, instead of expecting that all of `> 0`, `== 0`, and `< 0` will be false, in order to avoid `/fp:fast` shenanigans. Occurs for the imaginary part below.

This would be consistent with the beginning of `_Div` which tests for nan first.